### DISCUSSION TITLE
Recommended way to send emails from a Hummingbird application?

### DISCUSSION THREAD

**Discussion by ghost** [hummingbird-project/hummingbird#615] [Q&A]
Preferably without using an external service like SendGrid.

is https://github.com/sersoft-gmbh/swift-smtp a good choice? Or is there a preferred way to do this by the project team?

Thank you in advance for any help,

Mac

**Comment by adam-fowler** [comment]
I think quite a few people use https://github.com/Joannis/SMTPKitten. This was getting updated for structured concurrency but that worked got stalled. @Joannis is there any update on this?

**Comment by ghost** [comment]
@adam-fowler this also seems like a possibility in the near future: https://github.com/apple/swift-nio-imap

**Comment by adam-fowler** [comment]
> @adam-fowler this also seems like a possibility in the near future: https://github.com/apple/swift-nio-imap

IMAP is for retrieving email, not sending

**Comment by ghost** [comment]
> > @adam-fowler this also seems like a possibility in the near future: https://github.com/apple/swift-nio-imap
> 
> IMAP is for retrieving email, not sending

Ah my bad, I must have gotten confused.

**Comment by ghost** [comment]
> I think quite a few people use https://github.com/Joannis/SMTPKitten. This was getting updated for structured concurrency but that worked got stalled. @Joannis is there any update on this?

Another option is https://github.com/sersoft-gmbh/swift-smtp I guess I will test both and see which I prefer working with.

**Comment by adam-fowler** [comment]
Moving this to discussions
